public/app/features/dashboard/containers/DashboardPageProxy.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+
+import { config } from '@grafana/runtime';
+import { contextSrv } from 'app/core/core';
+import { GrafanaRouteComponentProps } from 'app/core/navigation/types';
+import DashboardScenePage from 'app/features/dashboard-scene/pages/DashboardScenePage';
+import { AccessControlAction } from 'app/types';
+
+import DashboardPage from './DashboardPage';
+import { DashboardPageRouteParams, DashboardPageRouteSearchParams } from './types';
+
+type Props = GrafanaRouteComponentProps<DashboardPageRouteParams, DashboardPageRouteSearchParams>;
+
+// This proxy component is used for Dashboard -> Scenes migration.
+// It will render DashboardScenePage if user does not have write permissions to a dashboard.
+function DashboardPageProxy(props: Props) {
+  if (config.featureToggles.dashboardSceneForViewers) {
+    if (contextSrv.hasPermission(AccessControlAction.DashboardsWrite)) {
```

Comment:
how does this hasPermission thing work? what if you have admin/write permission on folder or specific dashboard?

Reply:
I'm figuring this out. I think we gonna have to check permissions downstream anyways, not sure if the check here is sufficient.

Reply:
@torkelo I've changed this in the latest commit. The DashboardPageProxy ain't no longer there, but instead, it's the old "good" DashboardPage that will render the `DashboardScenePage` instead of its regular content. 

At the moment I think this is inevitable without a bigger refactor as the new page and the current one are using a different state management technique. And we need to get the dashboard first in order to see whether or not a user has permission to edit. Not an extremely nice approach I think but it's solving the problem without major changes to the old arch and without making unnecessary API calls to fetch the dashboard and read permissions.

Alt approach is to introduce a new dashboard API endpoint that would return dashboard meta. But it's not too appealing given we would use it for this temporary solution only.

Reply:
you are right. I've actually added a short-lived cache and I think for the case we are looking at this should be sufficient, as this is primarily proxying to a component. 

The cache lives on the `DashboardScenePageStateManager`. Could possibly be a separate instance, but I think it's better to keep it close to where eventually the sate will be managed. So dashboard loader now uses this cache to query the dashboard, and if not present or expiered it will hit the API.